server/src/__tests__/hover.test.ts

```diff
@@ -604,4 +604,51 @@ describe('on hover', () => {
       })
     )
   })
+
+  it('should show final value of the variable after the scan results are available', async () => {
```

Comment:
The commit message for "Feat: Show final values on hover for variables after the scan results… " should be renamed "Test: ..." because it only implements the test.